lightning/src/routing/router.rs

```diff
@@ -102,7 +102,7 @@ impl<G: Deref<Target = NetworkGraph<L>> + Clone, L: Deref, S: Deref, SP: Sized,
 			.filter(|details| details.counterparty.features.supports_route_blinding())
 			.filter(|details| amount_msats <= details.inbound_capacity_msat)
 			.filter(|details| amount_msats >= details.inbound_htlc_minimum_msat.unwrap_or(0))
-			.filter(|details| amount_msats <= details.inbound_htlc_maximum_msat.unwrap_or(0))
+			.filter(|details| amount_msats <= details.inbound_htlc_maximum_msat.unwrap_or(u64::MAX))
```

Comment:
Hmm... I think my reasoning here was that `inbound_htlc_maximum_msat` should be set for usable channels. So if it is not set, using zero would cause us to not consider this channel.

It will be `Some` if `ChannelContext::counterparty_selected_channel_reserve_satoshis` is `Some`, which seems to be when `accept_channel` is received:

https://github.com/lightningdevkit/rust-lightning/blob/b9797ebdd92a74436de6489bd1e1a6fb8990841e/lightning/src/ln/channel.rs#L6324